scalpel/src/Text/HTML/Scalpel.hs

```diff
@@ -66,33 +66,35 @@
 -- The following snippet defines a function, @allComments@, that will download
 -- the web page, and extract all of the comments into a list:
 --
--- > type Author = String
--- >
--- > data Comment
--- >     = TextComment Author String
--- >     | ImageComment Author URL
--- >     deriving (Show, Eq)
--- >
--- > allComments :: IO (Maybe [Comment])
--- > allComments = scrapeURL "http://example.com/article.html" comments
--- >    where
--- >        comments :: Scraper String [Comment]
--- >        comments = chroots ("div" @: [hasClass "container"]) comment
--- >
--- >        comment :: Scraper String Comment
--- >        comment = textComment <|> imageComment
--- >
--- >        textComment :: Scraper String Comment
--- >        textComment = do
--- >            author      <- text $ "span" @: [hasClass "author"]
--- >            commentText <- text $ "div"  @: [hasClass "text"]
--- >            return $ TextComment author commentText
--- >
--- >        imageComment :: Scraper String Comment
--- >        imageComment = do
--- >            author   <- text       $ "span" @: [hasClass "author"]
--- >            imageURL <- attr "src" $ "img"  @: [hasClass "image"]
--- >            return $ ImageComment author imageURL
+-- @
+-- type Author = String
+--
+-- data Comment
+--     = TextComment Author String
+--     | ImageComment Author URL
+--     deriving (Show, Eq)
+--
+-- allComments :: IO (Maybe [Comment])
+-- allComments = 'scrapeURL' \"http:\/\/example.com/article.html\" comments
+--    where
+--        comments :: Scraper String [Comment]
+--        comments = 'chroots' ("div" '@:' ['hasClass' "container"]) comment
+--
+--        comment :: Scraper String Comment
+--        comment = textComment `<|>` imageComment
+--
+--        textComment :: Scraper String Comment
+--        textComment = do
+--            author      <- 'text' $ "span" \@: [hasClass "author"]
```

Comment:
This breaks the formatting here and below.

Reply:
In the do blocks `$` operators were vertically aligned. Now their alignment is off and they seem randomly spaced.

Reply:
They are not randomly spaced; if you build the haddocks you will notice the `$` operators line up. This is because Haddock interprets the markup signs in order to hyperlink stuff and removes them in the final documentation.

Reply:
Sounds good!